controllers/dbinstance_controller.go

```diff
@@ -0,0 +1,322 @@
+	"github.com/kloeckner-i/db-operator/pkg/utils/kci"
+	"github.com/kloeckner-i/db-operator/pkg/utils/proxy"
+	"github.com/sirupsen/logrus"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"time"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kciv1alpha1 "github.com/kloeckner-i/db-operator/api/v1alpha1"
+)
+
+var (
+	dbInstancePhaseValidate    = "Valitating"
+	dbInstancePhaseCreate      = "Creating"
+	dbInstancePhaseBroadcast   = "Broadcasting"
+	dbInstancePhaseProxyCreate = "ProxyCreating"
+	dbInstancePhaseRunning     = "Running"
+)
+
+// DbInstanceReconciler reconciles a DbInstance object
+type DbInstanceReconciler struct {
+	client.Client
+	Log      logr.Logger
+	Scheme   *runtime.Scheme
+	Interval time.Duration
+	Conf     *config.Config
+}
+
+//+kubebuilder:rbac:groups=kci.rocks,resources=dbinstances,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=kci.rocks,resources=dbinstances/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=kci.rocks,resources=dbinstances/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the DbInstance object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.7.2/pkg/reconcile
+func (r *DbInstanceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = r.Log.WithValues("dbinstance", req.NamespacedName)
+
+	reconcilePeriod := r.Interval * time.Second
+	reconcileResult := reconcile.Result{RequeueAfter: reconcilePeriod}
+
+	// Fetch the DbInstance custom resource
+	dbin := &kciv1alpha1.DbInstance{}
+	err := r.Get(ctx, req.NamespacedName, dbin)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcileResult, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcileResult, err
+	}
+
+	// Update object status always when function returns, either normally or through a panic.
+	defer r.Status().Update(ctx, dbin)
+
+	// Check if spec changed
+	if isDBInstanceSpecChanged(dbin) {
+		logrus.Infof("Instance: name=%s spec changed", dbin.Name)
+		dbin.Status.Status = false
+		dbin.Status.Phase = dbInstancePhaseValidate // set phase to initial state
+	}
+
+	phase := dbin.Status.Phase
+	logrus.Infof("Instance: name=%s %s", dbin.Name, dbin.Status.Phase)
```

Comment:
phase used in logging in two different ways

https://github.com/kloeckner-i/db-operator/pull/60/files#diff-76309751b88102aab931e79aaf718bc2e4a2bb9580311be75b8c95593d20834eR144

maybe keep using var  `phase` after assignment or keep everywhere the object call.